nipype/interfaces/base/support.py

```diff
@@ -215,34 +196,68 @@ class InterfaceResult(object):
         * stderr : Any error messages output from running ``cmdline``.
         * returncode : The code returned from running the ``cmdline``.
 
+
+    >>> rt = InterfaceRuntime(cmdline='/bin/echo', returncode=0)
+    >>> result = InterfaceResult(interface='CommandLine',
+    ...                          runtime=rt)
+    >>> import pickle
+    >>> pickleds = pickle.dumps(result)
+    >>> newresult = pickle.loads(pickleds)
+    >>> newresult  # doctest: +ELLIPSIS
+    InterfaceResult(interface='CommandLine', runtime=\
+Bunch(cmdline='/bin/echo', returncode=0), version=...)
+
     """
+    __slots__ = ['interface', 'runtime', 'inputs', 'outputs', 'provenance']
+    version = __version__
 
     def __init__(self,
                  interface,
                  runtime,
                  inputs=None,
                  outputs=None,
                  provenance=None):
-        self._version = 2.0
         self.interface = interface
         self.runtime = runtime
         self.inputs = inputs
         self.outputs = outputs
         self.provenance = provenance
 
-    @property
-    def version(self):
-        return self._version
 
+    def __setstate__(self, state):
+        """Necessary for un-pickling"""
+        for key in self.__class__.__slots__:
+            setattr(self, key, state.get(key, None))
```

Comment:
Should you check that `state.get('version') == self.version`?

Reply:
Throw a warning at least would be appropriate. Since I was trying to do a minimal proof of concept, I passed. But yes, I'll include the warning.

Reply:
Warning added :+1: